src/wp-includes/media.php

```diff
@@ -4767,6 +4767,95 @@ function get_post_galleries( $post, $html = true ) {
 		}
 	}
 
+	if ( has_block( 'gallery', $post->post_content ) ) {
+		$post_blocks = parse_blocks( $post->post_content );
+
+		while ( $block = array_shift( $post_blocks ) ) {
+			$has_inner_blocks = ! empty( $block['innerBlocks'] );
+
+			// Skip blocks with no blockName and no innerHTML.
+			if ( ! $block['blockName'] ) {
+				continue;
+			}
+
+			// All blocks nested inside non-Gallery blocks should be in the root array.
+			if ( $has_inner_blocks && 'core/gallery' !== $block['blockName'] ) {
+				array_push( $post_blocks, ...$block['innerBlocks'] );
+				continue;
+			}
+
+			// New Gallery block format as HTML.
+			if ( $has_inner_blocks && $html ) {
+				$block_html  = wp_list_pluck( $block['innerBlocks'], 'innerHTML' );
+				$galleries[] = '<figure>' . implode( ' ', $block_html ) . '</figure>';
+				continue;
+			}
+
+			$srcs = array();
+
+			// New Gallery block format as an array.
+			if ( $has_inner_blocks ) {
+				$attrs = wp_list_pluck( $block['innerBlocks'], 'attrs' );
+				$ids   = wp_list_pluck( $attrs, 'id' );
+
+				foreach ( $ids as $id ) {
+					$url = wp_get_attachment_url( $id );
+
+					if ( $url && ! in_array( $url, $srcs, true ) ) {
```

Comment:
`wp_get_attachment_url()` returns a string if URL found, else `false` (though filtered). As it's filtered, could be anything coming back including something that passes the truthy check.

@costdev what about changing this to `is_string( $src )` to ensure it's a string data type?

```suggestion
					if ( is_string( $url ) && ! in_array( $url, $srcs, true ) ) {
```

Here's the benchmark on truthy vs `is_string()` https://3v4l.org/IGHWg#v7.4.25. Truthy is slightly faster on average at 0.0028 μs. But the `is_string()` is more clear and guards the `in_array()` to do its thang only if a string URL is found. Thoughts?

Reply:
Sounds great, and the speed difference is minimal to bring a better check. Interestingly, `false !== $url` is slower than both of them. I'll add `is_string( $url )` to the next iteration.